.github/workflows/tests.yml

```
@@ -494,6 +494,231 @@ jobs:
+          export CGO_CFLAGS="-I/home/runner/work/lxd/lxd-test/vendor/dqlite/include/"
+          export CGO_LDFLAGS="-L/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LD_LIBRARY_PATH="/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LXD_DOCUMENTATION="/home/runner/work/lxd/lxd/doc/_build/"
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo rm -rf /var/lib/lxd
+          sudo -E PATH=${PATH} LD_LIBRARY_PATH=${LD_LIBRARY_PATH} $(go env GOPATH)/bin/lxd --group sudo &
+
+      - name: Setup LXD
+        shell: bash
+        run: |
+          set -eux
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo -E LXD_DIR=/var/lib/lxd lxc storage create default zfs
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default root disk path=/ pool=default
+          sudo -E LXD_DIR=/var/lib/lxd lxc network create local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default eth0 nic network=local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set core.https_address "[::]:8443"
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set cluster.https_address "127.0.0.1"
+          sudo -E LXD_DIR=/var/lib/lxd lxc cluster enable local
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set user.show_permissions=true
+
+      - name: Checkout LXD-UI
+        uses: actions/checkout@v4
+        with:
+          repository: 'canonical/lxd-ui'
+          ref: main
+          path: lxd-ui
+
+      - name: Install Dotrun
+        run: sudo pip3 install dotrun
+
+      - name: Restore cached keys
+        uses: actions/cache/restore@v3
+        with:
+          path: lxd-ui/keys
+          key: keys-folder
+
+      - name: Install LXD-UI dependencies
+        run: |
+          set -eux
+          sudo chmod 0777 ./lxd-ui
+          cd lxd-ui
+          dotrun install
+
+      - name: Run LXD-UI
+        env:
+          ENVIRONMENT: devel
+          PORT: 8407
+          LXD_UI_BACKEND_IP: 172.17.0.1
+        run: |
+          set -eux
+          cd lxd-ui
+          dotrun &
+          curl --head --fail --retry-delay 2 --retry 100 --retry-connrefused --insecure https://localhost:8407
+
+      - name: Set keys permissions
+        run: |
+          set -eux
+          sudo chmod -R 0666 lxd-ui/keys
+          sudo chmod 0777 lxd-ui/keys
+
+      - name: Save keys
+        uses: actions/cache/save@v3
+        with:
+          path: lxd-ui/keys
+          key: keys-folder
+
+      - name: Allow LXD-UI keys
+        shell: bash
+        run: |
+          set -eux
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo -E LXD_DIR=/var/lib/lxd lxc config trust add lxd-ui/keys/lxd-ui.crt
+
+      - name: Create a custom image
+        shell: bash
+        run: |
+          set -eux
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo -E LXD_DIR=/var/lib/lxd lxc launch ubuntu-minimal:22.04 my-instance
```

No need to launch, init will do. Also, we tend to use `minimal-daily` elsewhere:
```suggestion
          sudo -E LXD_DIR=/var/lib/lxd lxc init ubuntu-minimal-daily:22.04 my-instance
```

But if you don't actually need a full fledged instance, we have a simple busybox `testimage`. See `test/includes/setup.sh`.

---

This is a workaround to generate an image from an instance, before this was not possible in the UI. I think by now we can generate the image directly in our e2e test. So we might be able to get rid of this preparation step completely. I need to double-check if we can evolve our test, but I am optimistic it will work. Will keep this open for now.

---

removed this bit after adjusting the ui tests.